tdns/agent_utils.go

```
@@ -0,0 +1,612 @@
+						svcname = fmt.Sprintf("_https._tcp.%s", remoteid)
+					}
+					svcname = dns.Fqdn(svcname)
+
+					rrset, err := RecursiveDNSQueryWithServers(svcname, dns.TypeURI, timeout, retries, resolvers)
+					if err != nil {
+						log.Printf("LocateAgent: error response to URI query for %s transport: %v", transport, err)
+						continue // Try the next transport instead of failing
+					}
+
+					// Process URI response
+					for _, rr := range rrset.RRs {
```

golangci-lint (1.62.2)

SA5011: possible nil pointer dereference

(staticcheck)